rosette/src/Vm.cc

```diff
@@ -503,7 +497,16 @@ void VirtualMachine::handleSleep() {
 }
 
 void VirtualMachine::handleSignal() {
-    int oldmask = sigblock(IO_MASK);
+    sigset_t oldmask;
+    sigset_t blockmask;
+
+    sigemptyset(&blockmask);
+    sigaddset(&blockmask, SIGIO);
+    if (sigprocmask(SIG_BLOCK, &blockmask, &oldmask) < 0) {
+        printf("aborting...\n");
```

Comment:
`fprintf(stderr, "...")` perhaps? Also, "aborting..." is a pretty lame error message unless something was being printed before that? Can you upgrade the error?